### PR TITLE
Top 5

### PR DESCRIPTION
**Pull request by jaw772**
- Creation of factories to create multiple transactions for merchants and customers
- top_customers method reconstruction to filter for specifically transactions with the success result
- Both user story 38 and 20 were completed in this
- Driver: Josh / Navigator: Leland

### CODE REVIEW

File: spec/features/admin/dashboard_spec.rb [hunk 1]
```diff
@@ -21,20 +21,19 @@
     customer_4 = create(:customer, first_name: 'Zach', last_name: "Doe")
     customer_5 = create(:customer, first_name: 'Charlie', last_name: "Rey")
 
-    merchant_1 = create(:merchant_with_invoices, invoice_count: 6, customer: customer_1, invoice_status: 2)
-    merchant_2 = create(:merchant_with_invoices, invoice_count: 3, customer: customer_2, invoice_status: 2)
-    merchant_3 = create(:merchant_with_invoices, invoice_count: 8, customer: customer_3, invoice_status: 2)
-    merchant_4 = create(:merchant_with_invoices, invoice_count: 1, customer: customer_4, invoice_status: 2)
-    merchant_5 = create(:merchant_with_invoices, invoice_count: 4, customer: customer_5, invoice_status: 2)
-
+    merchant_1 = create(:merchant_with_transactions, transaction_count: 6, customer: customer_1, transaction_result: 0)
```
Comment (LelandCurtis) [inline spec/features/admin/dashboard_spec.rb:24]: This is only creating one customer per merchant. Is this what you want? Do we need to create a second customer for one of the merchants?

**Review by LelandCurtis — APPROVED** [review]
- looks good. Did you intend to only create 1 customer per merchant for the admin/merchant page top customers? Is this testing top customers across all merchants?

**Review by pleonar1 — COMMENTED** [review]
I think It looks good

File: spec/features/admin/dashboard_spec.rb (hunk 1 above)
Comment (jaw772) [reply spec/features/admin/dashboard_spec.rb:24]: I just wanted to have different merchants. it's not required for the user story. 

EDIT: Dylan just posted in the code tips channel that we need to test for multiple merchants.

**Pull request merged**